src/engine/Join.cpp

```diff
@@ -762,77 +763,56 @@ void Join::hashJoin(const IdTable& dynA, size_t jc1, const IdTable& dynB,
 
   IdTableStatic<OUT_WIDTH> result = dynRes->moveToStatic<OUT_WIDTH>();
 
-  // Cannot just switch l1 and l2 around because the order of
+  // Puts the rows of the given table into a hash map, with the value of
+  // the join column of a row as the key, and returns the hash map.
+  auto putIntoHashMap = [] (const auto table, const size_t jc, const auto WIDTH) {
+    // This declaration works, because generic lambdas are just syntactic sugar
+    // for templates. So, if the parameter value of WIDTH is avaible at
+    // compilet ime (and marked as such with std::integral_constant), we can
+    // use WIDTH for a template parameter.
+    ad_utility::HashMap<Id, std::vector<std::decay_t<typename IdTableStatic<WIDTH>::const_row_type>>> map;
```

Comment:
You can infer the row type directly from the `
```suggestion
    ad_utility::HashMap<Id, std::vector<std::decay_t<typename Table::const_row_type>>> map;
```
(no more explicit WIDTH needed)

Reply:
Done. See [88e172c](https://github.com/ad-freiburg/qlever/pull/821/commits/88e172c9caec3fdc591de766d50861acdaad4a94).